### PR TITLE
Use upstream RunKubelet method to start kubelet

### PR DESCRIPTION
**Pull request by csrwng**
Switch to upstream code to start the kubelet.

### CODE REVIEW

**Comment by csrwng** [comment]
@smarterclayton ptal

File: pkg/cmd/server/kubernetes/node.go [hunk 1]
```diff
@@ -189,73 +157,76 @@ func (c *NodeConfig) RunKubelet() {
...
+		AllowPrivileged:                true,
+		HostNetworkSources:             hostNetworkCapabilities,
+		HostnameOverride:               c.NodeHost,
+		RootDirectory:                  c.VolumeDir,
+		ConfigFile:                     c.PodManifestPath,
+		ManifestURL:                    "",
+		FileCheckFrequency:             time.Duration(c.PodManifestCheckIntervalSeconds) * time.Second,
+		HTTPCheckFrequency:             0,
+		PodInfraContainerImage:         c.ImageFor("pod"),
+		SyncFrequency:                  3 * time.Second,
```
Comment (csrwng) [inline pkg/cmd/server/kubernetes/node.go]: @smarterclayton - The upstream interval is 10 sec. I kept what we currently have. Is that intentional?

**Comment by csrwng** [comment]
[test]

**Comment by csrwng** [comment]
Opened issue https://github.com/GoogleCloudPlatform/kubernetes/issues/8424 for recurring message about cadvisor.

File: pkg/cmd/server/kubernetes/node.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/cmd/server/kubernetes/node.go]: Hrm - maybe it should be shorter.

Did you see my changes in https://github.com/GoogleCloudPlatform/kubernetes/pull/8386?

Comment (csrwng) [inline pkg/cmd/server/kubernetes/node.go]: I did now. When we pull that change we'll need to check for an error here: https://github.com/openshift/origin/pull/2323/files#diff-5009334e857b25f4efb55d2d21f157e1R229 which is good.

Comment (smarterclayton) [inline pkg/cmd/server/kubernetes/node.go]: Doesn't this method block on exit?

Comment (csrwng) [inline pkg/cmd/server/kubernetes/node.go]: If we get to this point  https://github.com/smarterclayton/kubernetes/commit/dc0e3dd0a474f63f242226349c525f1a7465013f#diff-e0160f895346669c2e4495acfd858ea7R365, yes. But now you return an error if something bad happens before that, so we'll need to handle it.

Comment (csrwng) [inline pkg/cmd/server/kubernetes/node.go]: BTW, the reason I asked about the sync interval is that currently container garbage collection is set to run every minute (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/kubelet.go#L633). If the kubelet syncs every 3 secs, you can get 20 containers per failed pod before they are garbage collected.

Comment (smarterclayton) [inline pkg/cmd/server/kubernetes/node.go]: The point is we didn't block here before, but now we do.  So we can't block.

Comment (smarterclayton) [inline pkg/cmd/server/kubernetes/node.go]: Our sync interval should be whatever upstream uses if it's longer than what we use now.

**Comment by csrwng** [comment]
@smarterclayton - is this ok to be merged? It was failing jenkins because I was passing the wrong flag for EnableDebuggingHandlers.

**Comment by csrwng** [comment]
@smarterclayton - I looked a little bit more into this code and I don't see where it blocks. https://github.com/openshift/origin/pull/2323/files#diff-5009334e857b25f4efb55d2d21f157e1R229 returns after the kubelet go routines have been kicked off: https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/cmd/kubelet/app/server.go#L547 and https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/cmd/kubelet/app/server.go#L552.

**Comment by smarterclayton** [comment]
Rebase merged, can you rebase this and I'll merge?

**Comment by csrwng** [comment]
sure

**Comment by csrwng** [comment]
rebased

File: pkg/cmd/server/kubernetes/node.go [hunk 2]
```diff
@@ -144,42 +142,12 @@ func (c *NodeConfig) RunKubelet() {
...
 
 	cadvisorInterface, err := cadvisor.New(4194)
-	if err == nil {
-		// TODO: use VersionInfo after the next rebase
-		_, err = cadvisorInterface.MachineInfo()
-	}
 	if err != nil {
-		glog.Errorf("WARNING: cAdvisor cannot be started: %v", err)
-		cadvisorInterface = &cadvisor.Fake{}
+		glog.Fatalf("Error instantiating cadvisor: %v", err)
```
Comment (smarterclayton) [inline pkg/cmd/server/kubernetes/node.go]: Does this still return unsupported on macs?

**Comment by smarterclayton** [comment]
[merge]

File: pkg/cmd/server/kubernetes/node.go (hunk 2 above)
Comment (csrwng) [inline pkg/cmd/server/kubernetes/node.go]: It does when I cross-compile to linux natively on the mac. I'm trying with a binary built with the openshift-release container to see if it makes a difference. (Just much slower)

Comment (csrwng) [inline pkg/cmd/server/kubernetes/node.go]: It works fine in boot2docker as long as I build the binary with the release container (which I guess is the official way of building it anyway). It's just slower to build than a native cross-compile.

**Pull request merged**